passes/cmds/rename.cc

```diff
@@ -210,114 +210,102 @@ struct RenamePass : public Pass {
 		{
 			extra_args(args, argidx, design);
 
-			for (auto &mod : design->modules_)
+			for (auto module : design->selected_modules())
 			{
 				int counter = 0;
+				std::map<RTLIL::Wire *, IdString> new_wire_names;
+				std::map<RTLIL::Cell *, IdString> new_cell_names;
+
+				for (auto wire : module->selected_wires())
+					if (wire->name[0] == '$')
+						new_wire_names[wire] = derive_name_from_src(wire->get_src_attribute(), counter++);
+
+				for (auto cell : module->selected_cells())
+					if (cell->name[0] == '$')
+						new_cell_names[cell] = derive_name_from_src(cell->get_src_attribute(), counter++);
 
-				RTLIL::Module *module = mod.second;
-				if (!design->selected(module))
-					continue;
-
-				dict<RTLIL::IdString, RTLIL::Wire*> new_wires;
-				for (auto &it : module->wires_) {
-					if (it.first[0] == '$' && design->selected(module, it.second))
-						it.second->name = derive_name_from_src(it.second->get_src_attribute(), counter++);
-					new_wires[it.second->name] = it.second;
-				}
-				module->wires_.swap(new_wires);
-				module->fixup_ports();
-
-				dict<RTLIL::IdString, RTLIL::Cell*> new_cells;
-				for (auto &it : module->cells_) {
-					if (it.first[0] == '$' && design->selected(module, it.second))
-						it.second->name = derive_name_from_src(it.second->get_src_attribute(), counter++);
-					new_cells[it.second->name] = it.second;
-				}
-				module->cells_.swap(new_cells);
+				for (auto &it : new_wire_names)
+					module->rename(it.first, it.second);
+
+				for (auto &it : new_cell_names)
+					module->rename(it.first, it.second);
 			}
 		}
 		else
 		if (flag_wire)
 		{
 			extra_args(args, argidx, design);
 
-			for (auto &mod : design->modules_)
-			{
-				RTLIL::Module *module = mod.second;
-				if (!design->selected(module))
-					continue;
-
-				dict<RTLIL::IdString, RTLIL::Cell*> new_cells;
-				for (auto &it : module->cells_) {
-					if (it.first[0] == '$' && design->selected(module, it.second))
-						it.second->name = derive_name_from_wire(*it.second);
-					new_cells[it.second->name] = it.second;
-				}
-				module->cells_.swap(new_cells);
+			for (auto module : design->selected_modules()) {
+				std::map<RTLIL::Cell *, IdString> new_cell_names;
+				for (auto cell : module->selected_cells())
+					if (cell->name[0] == '$')
+						new_cell_names[cell] = derive_name_from_cell_output_wire(cell);
+				for (auto &it : new_cell_names)
+					module->rename(it.first, it.second);
 			}
 		}
 		else
 		if (flag_enumerate)
 		{
 			extra_args(args, argidx, design);
 
-			for (auto &mod : design->modules_)
+			for (auto module : design->selected_modules())
 			{
 				int counter = 0;
+				std::map<RTLIL::Wire *, IdString> new_wire_names;
+				std::map<RTLIL::Cell *, IdString> new_cell_names;
+
+				for (auto wire : module->selected_wires())
+					if (wire->name[0] == '$') {
+						std::ostringstream buf;
```

Comment:
I don't think using `ostringstream` is an improvement in readability over `stringf`. STL iostreams have a notoriously baroque interface, which is particularly bad when you need to use any formatting options, whereas printf-style functions don't have that issue. It is also more verbose.

Reply:
TBH I don't mind either way, I was just updating to use C++ stdlib instead of libc functions.

Reply:
Even though it looks like one, `stringf` isn't actually a libc function: it's defined in `yosys.h`. We prefer it over stringstreams so please don't convert use of `stringf`!

Reply:
Oh, I see.  My bad :)